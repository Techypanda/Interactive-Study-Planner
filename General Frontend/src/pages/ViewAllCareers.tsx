// careers display with a search. Somehow this links to the course selection thingy
// don't call this yet it's not even close to done
import { BrowserRouter as Router } from "react-router-dom";
import { Box, Button, Grid, Typography, Card, CardHeader, TextField, Container} from "@material-ui/core";
import Navbar from "../components/shared/Navbar";
import { useHistory } from 'react-router-dom';
import styled from "styled-components";
import { useQuery, useQueryClient } from "react-query";
import axios, { AxiosError, AxiosResponse } from "axios";
import { DefaultProps } from "../types";
import CareerEntry from "../components/shared/CareerEntry";

// called on loading of this component
// makes call to the careers API to get all careers so they can be rendered
function ListCareers(props: DefaultProps) {
  const { isLoading, isError, error, data} = useQuery('getcareers', async() => {
	  const data = await axios('${process.env.REACT_APP_API_URI}/event-get-all-careers');
	  return data;
  })
  if (isLoading) {
	  return "Retrieving all careers";
  }
  if (isError) {
	  return error;
  }
  return (
    <>
      <Navbar/>
      <Container id="listcareers" className={props.className}>
        <Box id="searchcontainer" display="flex">
          <TextField variant="outlined" placeholder="Search careers..." className="searchbar" />
          <Button color="primary" variant='contained' className="searchbtn">GO</Button>
        </Box>
        <Box id="careers" marginTop={2}>
          <CareerEntry careerTitle="idk" careerContent="wagecuckery"/>
        </Box>
      </Container>
    </>
  );
}

export default /*styled(ListCareers)`
  #titlebar h4 {
    margin: auto;
  }
  #searchcontainer .searchbtn {
    margin-left: 15px;
  }
  #searchcontainer .searchbar {
    height: 36px;
    flex-grow: 1;
  }
  #searchcontainer .searchbar .MuiInputBase-root {
    height: 100%;
  }
`*/ListCareers;
