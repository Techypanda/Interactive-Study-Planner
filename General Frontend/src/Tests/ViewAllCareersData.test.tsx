import React from 'react';
import { render } from '@testing-library/react';
import ViewAllCareers from '../pages/ViewAllCareers';

describe('View All Careers is working which happens When', () => {
    it('Renders Without A Crash', () => {
        render(<ViewAllCareers />)
    })
    it('Renders And Matches Snapshot', () => {
        const x = render(<ViewAllCareers />)
        expect(x).toMatchSnapshot();
    })
})