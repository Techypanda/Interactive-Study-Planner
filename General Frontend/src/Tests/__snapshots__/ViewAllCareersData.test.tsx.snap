// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View All Careers is working which happens When Renders And Matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-bdnxRM hKTgfL MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular flexit MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-19"
            id="LogoSection"
          >
            <img
              alt="curtin logo"
              src="curtinlogo.png"
            />
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            <b>
              Medical Course Planner  
            </b>
          </h6>
        </div>
      </header>
      <br />
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-20"
          id="searchcontainer"
        >
          <div
            class="MuiFormControl-root MuiTextField-root searchbar MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="standard-full-width"
                placeholder="Search careers..."
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-21 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-22"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained searchbtn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <br />
        <div
          class="career-option"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-outlined MuiPaper-rounded"
          >
            <div
              class="makeStyles-details-14"
            >
              <div
                class="MuiCardContent-root makeStyles-content-15"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
                />
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
                />
              </div>
              <div
                class="makeStyles-controls-17"
              />
            </div>
          </div>
        </div>
      </div>
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-bdnxRM hKTgfL MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular flexit MuiToolbar-gutters"
      >
        <div
          class="MuiBox-root MuiBox-root-19"
          id="LogoSection"
        >
          <img
            alt="curtin logo"
            src="curtinlogo.png"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          <b>
            Medical Course Planner  
          </b>
        </h6>
      </div>
    </header>
    <br />
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-20"
        id="searchcontainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root searchbar MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="standard-full-width"
              placeholder="Search careers..."
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-21 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-22"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained searchbtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <br />
      <div
        class="career-option"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="makeStyles-details-14"
          >
            <div
              class="MuiCardContent-root makeStyles-content-15"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
              />
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
              />
            </div>
            <div
              class="makeStyles-controls-17"
            />
          </div>
        </div>
      </div>
    </div>
    <br />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
